src/sys/mman.rs

```diff
@@ -519,8 +518,8 @@ pub unsafe fn mremap(
 /// page.
 ///
 /// [`munmap(2)`]: https://man7.org/linux/man-pages/man2/munmap.2.html
-pub unsafe fn munmap(addr: *mut c_void, len: size_t) -> Result<()> {
-    Errno::result(libc::munmap(addr, len)).map(drop)
+pub unsafe fn munmap(mut addr: NonNull<c_void>, len: size_t) -> Result<()> {
+    Errno::result(libc::munmap(addr.as_mut(), len)).map(drop)
```

Comment:
Why we are using `NonNull.as_mut()` for `munmap()`? Other functions are all using `NonNull.as_ptr()`

Reply:
Added comment.